nifi-extension-bundles/nifi-github-bundle/nifi-github-extensions/src/main/java/org/apache/nifi/github/GitHubRepositoryClient.java

```diff
@@ -379,6 +392,41 @@ public GHContent deleteContent(final String filePath, final String commitMessage
         });
     }
 
+    /**
+     * Creates the JwtToken for Authentication with Private Key
+     *
+     * @param pemString is the PKCS#1 String
+     * @return the JwtToken
+     *
+     * @throws Exception if an error occurs parsing key
+     */
+    private static String loadPrivateKeyFromPEM(String pemString) throws Exception {
+        long nowMillis = System.currentTimeMillis();
+        long expMillis = nowMillis + 600000; // Token validity 10 minutes
+        Date now = new Date(nowMillis);
+        Date exp = new Date(expMillis);
+        PEMParser pemParser = new PEMParser(new StringReader(pemString));
+        Object object = pemParser.readObject();
+        pemParser.close();
+
+        if (object instanceof PEMKeyPair) {
+            PEMKeyPair keyPair = (PEMKeyPair) object;
+            RSAPrivateKey rsaPrivateKey = RSAPrivateKey.getInstance(keyPair.getPrivateKeyInfo().parsePrivateKey());
+            PKCS8EncodedKeySpec keySpec = new PKCS8EncodedKeySpec(rsaPrivateKey.getEncoded());
```

Comment:
We should use the utility from the github-api library, even though it is limited to PKCS 8. This avoids the use of the Bouncy Castle library, and also avoids the custom methods for token generation.

Reply:
I'll commit the changes you requested and will use the utility function for authentication.

Reply:
I removed the seperate function for jwt and used JWTTokenProvider for token generation.